dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/TeradataLogsJdbcTask.java

```diff
@@ -171,17 +176,21 @@ private String getSql(@Nonnull JdbcHandle handle) {
    * @param predicate A predicate to compute whether a given expression is legal.
    * @return A SQL query containing every legal expression from EXPRESSIONS.
    */
-  @ForOverride
   @Nonnull
-  /* pp */ String getSql(@Nonnull Predicate<? super String> predicate) {
-    return getSql(predicate, EXPRESSIONS);
+  /* pp */ String getSql(
```

Comment:
optional: `getSql` method modifies class property now, so this is no longer a simple getter. I would suggest renaming it to sth more appropriate (e.g. prepareSql ?). 
Additionally if you think calling `buildSql` function twice is fine (once to get SQL, second to print sql) you could remove private `sql` property and simplify this logic.

Reply:
I've renamed the method to `getOrCreateSql`.